js/script.js

```diff
@@ -0,0 +1,161 @@
+    homeStreet: document.getElementById('homeStreet'),
+    homeCity: document.getElementById('homeCity'),
+    homeState: document.getElementById('homeState'),
+    homePhone: document.getElementById('homePhone'),
+    homePin: document.getElementById('homePin'),
+    homeFax: document.getElementById('homeFax'),
+    officeStreet: document.getElementById('officeStreet'),
+    officeCity: document.getElementById('officeCity'),
+    officeState: document.getElementById('officeState'),
+    officePhone: document.getElementById('officePhone'),
+    officePin: document.getElementById('officePin'),
+    officeFax: document.getElementById('officeFax'),
+    inputElements: document.getElementsByTagName('input'),
+    nameChecker: ['firstName', 'lastName', 'homeCity', 'homeState', 'officeState',
+        'employer', 'employment', 'officeCity'],
+    phoneChecker: ['homePhone', 'homeFax', 'officePhone', 'officeFax'],
+    addressChecker: ['homeStreet', 'officeStreet'],
+    pinChecker: ['homePin', 'officePin'],
+    errorSpan: document.createElement('span'),
+
+    validateName: function (nameInput) {
+        return this.regexName.exec(nameInput.value);
+    },
+    validatePhone: function (phoneInput) {
+        return this.regexPhone.exec(phoneInput.value);
+    },
+    validatePin: function (pinInput) {
+        return this.regexPin.exec(pinInput.value);
+    },
+    validateAddress: function (addressInput) {
+        return this.regexAddress.exec(addressInput.value);
+    },
+    validateDOB: function () {
+        return this.dob.value < this.currentDate;
+    },
+    addErrorclass: function (currentParentElement) {
+        currentParentElement.classList.add('has-feedback');
+        currentParentElement.classList.add('has-error');
+        $('<span class="glyphicon glyphicon-remove form-control-feedback" aria-hidden="true"></span>').appendTo(currentParentElement);
+        this.flag += 1;
+    },
+    removeErrorClass: function (currentParentElement) {
+        currentParentElement.classList.remove('has-feedback');
+        currentParentElement.classList.remove('has-error');
+        currentParentElement.removeChild(currentParentElement.childNodes[currentParentElement.childNodes.length - 1]);
+        this.flag -= 1;
+    },
+    checkAddress: function () {
+        if (this.checkboxAddress.checked) {
+            document.getElementById('officeAddress').disabled = true;
+            this.officeStreet.value = this.homeStreet.value;
+            this.officeCity.value = this.homeCity.value;
+            this.officeState.value = this.homeState.value;
+            this.officePin.value = this.homePin.value;
+            this.officePhone.value = this.homePhone.value;
+            this.officeFax.value = this.homeFax.value;
+        }
+        else {
+            document.getElementById('officeAddress').disabled = false;
+        }
+    },
+    validateInputElements: function () {
+
+        for (var i = 0; i < this.inputElements.length; i++) {
```

Comment:
suppose this.inputElements.length value is 1000, then every time for each counter i++ in for loop, it will calculate length of this.inputElements.
so its better to do it once that is before for loop and store in a variable and then use that variable in the for loop.